task-sdk/src/airflow/sdk/execution_time/supervisor.py

```diff
@@ -837,6 +839,15 @@ def wait(self) -> int:
         # If it hasn't, assume it's failed
         self._exit_code = self._exit_code if self._exit_code is not None else 1
 
+        self.update_task_state_if_needed()
+
+        # Now at the last possible moment, when all logs and comms with the subprocess has finished, lets
+        # upload the remote logs
+        self._upload_logs()
```

Comment:
Can we do this in finalize instead?

Reply:
That moves it to the child process (Task Runner) instead of Supervisor process

Reply:
I see, thanks